libs/MobileSync/MobileSync/Classes/Target/BriefcaseSyncDownTarget.swift

```diff
@@ -0,0 +1,349 @@
+//  WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+//  WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import Foundation
+import SalesforceSDKCore
+import Combine
+
+struct TypedId: Hashable {
+    let id: String
+    let objectInfo: BriefcaseObjectInfo
+}
+
+struct TypedIds {
+    var ids: [TypedId]
+    var objectTypeToIds: [BriefcaseObjectInfo: [String]] {
+        return Dictionary(grouping: ids) { $0.objectInfo }
+            .mapValues { $0.map { $0.id } }
+        
+    }
+    func sliceCount(sliceSize: Int) -> Int {
+        return Int((Double(ids.count) / Double(sliceSize)).rounded(.up))
+    }
+    
+    func slice(sliceIndex: Int, sliceSize: Int) -> TypedIds {
+        let minimumIndex = sliceIndex*sliceSize
+        let maxIndex = min(minimumIndex + sliceSize, ids.count)
+        return TypedIds(Array(ids[minimumIndex..<maxIndex]))
+    }
+
+    init(_ ids: [TypedId]) {
+        self.ids = ids
+    }
+}
+
+enum BriefcaseSyncDownError: Error {
+    case unknownResponse
+}
+
+@objc(SFBriefcaseSyncDownTarget)
+public class BriefcaseSyncDownTarget: SyncDownTarget {
+    private static let pageSize: UInt = 2000
+    private static let countIdsPerRetrieve = "countIdsPerRetrieve"
+    private static let infos = "infos"
+    private static let maxCountIdsPerRetrieve = 2000
+    
+    private var relayToken: String? = nil
+    private var maxTimeStamp: Int64 = 0
+    private var infosMap: [String: BriefcaseObjectInfo] = [:]
+    private var fetchedTypedIds: TypedIds?
+    private let countIdsPerRetrieve: Int
+    private var sliceIndex = 0
+    private var cancellableSet: Set<AnyCancellable> = []
+
+    override required public init(dict: [AnyHashable : Any]) {
+        if let infos = dict[BriefcaseSyncDownTarget.infos]  {
+            do {
+                let json = try JSONSerialization.data(withJSONObject: infos)
+                infosMap = try JSONDecoder().decode([String: BriefcaseObjectInfo].self, from: json)
+            } catch {
+                SalesforceLogger.log(BriefcaseSyncDownTarget.self, level: .error, message: "Error decoding briefcase info: \(error)")
+            }
+        }
+        
+        countIdsPerRetrieve = dict[BriefcaseSyncDownTarget.countIdsPerRetrieve] as? Int ?? BriefcaseSyncDownTarget.maxCountIdsPerRetrieve
+        super.init(dict: dict)
+        
+        self.queryType = QueryType.briefcase
+        ParentChildrenSyncHelper.registerAppFeature()
+    }
+    
+    @objc public convenience init(infos: [BriefcaseObjectInfo]) {
+        self.init(infos: infos, countIdsPerRetrieve: BriefcaseSyncDownTarget.maxCountIdsPerRetrieve)
+    }
+    
+    internal init(infos: [BriefcaseObjectInfo], countIdsPerRetrieve: Int) {
+        for info in infos {
+            infosMap[info.sobjectType] = info
+        }
+        self.countIdsPerRetrieve = countIdsPerRetrieve
+        ParentChildrenSyncHelper.registerAppFeature()
```

Comment:
sorry if I'm confused, I used this one on iOS because it was "RR" like Android, but looking again, I see a briefcase flag [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/0ebe5e33274ca66dd47513bfa1d9f7d781358708/libs/MobileSync/src/com/salesforce/androidsdk/mobilesync/app/Features.java#L37) that's not used yet, should I use "BC"?

Reply:
The code on Android was wrong (and I did not even realize it while copying from it for my comment above). It's being corrected with: https://github.com/forcedotcom/SalesforceMobileSDK-Android/pull/2316

Reply:
(I just corrected the comment above as well)